api-gateway/src/main/java/org/swasth/apigateway/models/ErrorRequest.java

```diff
@@ -0,0 +1,79 @@
+package org.swasth.apigateway.models;
+
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.swasth.apigateway.constants.Constants;
+import org.swasth.apigateway.exception.ClientException;
+import org.swasth.apigateway.exception.ErrorCodes;
+import org.swasth.apigateway.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.swasth.apigateway.constants.Constants.*;
+
+public class ErrorRequest {
+
+    private Map<String, Object> errorRequest;
+
+    public ErrorRequest(Map<String, Object> body) throws ClientException {
+        try {
+            this.errorRequest = body;
+        } catch (Exception e) {
+            throw new ClientException(ErrorCodes.ERR_INVALID_PAYLOAD, "Invalid payload");
+        }
+    }
+
+    public void validate(List<String> mandatoryHeaders,Map<String, Object> senderDetails, Map<String, Object> recipientDetails, String subject) throws ClientException {
+        List<String> missingHeaders = mandatoryHeaders.stream().filter(key -> !errorRequest.containsKey(key)).collect(Collectors.toList());
+        if (!missingHeaders.isEmpty()) {
+            throw new ClientException(ErrorCodes.ERR_MANDATORY_HEADERFIELD_MISSING, "Mandatory headers are missing: " + missingHeaders);
+        }
+
+
+        String correlationId = (String) errorRequest.get(Constants.CORRELATION_ID);
```

Comment:
Implement getCorrelationId as a method in pojo.

Reply:
Done